code/modules/mob/living/say.dm

```diff
@@ -106,45 +106,42 @@ var/list/department_radio_keys = list(
 				verb = "lisps"
 
 		if(T)
-			var/list/hearturfs = list()
-
-			for(var/I in hear(message_range, T))
-				if(istype(I, /mob/))
-					var/mob/M = I
-					listening += M
-					hearturfs += M.locs[1]
-					for(var/obj/O in M.contents)
-						if(O.flags_atom & USES_HEARING)
-							listening_obj |= O
-				else if(istype(I, /obj/))
-					var/obj/O = I
-					hearturfs += O.locs[1]
-					if(O.flags_atom & USES_HEARING)
-						listening_obj |= O
-
-			for(var/mob/M as anything in GLOB.player_list)
-				if((M.stat == DEAD || isobserver(M)) && M.client && M.client.prefs && (M.client.prefs.toggles_chat & CHAT_GHOSTEARS))
-					listening |= M
-					continue
-				if(M.loc && (M.locs[1] in hearturfs))
-					listening |= M
+			for(var/mob/M as anything in hearers(message_range, T))
+				listening += M
+				hearturfs += M.locs[1]
```

Comment:
Since we're primarily interested in looking up values, not iterating on every entry we enter into hearturfs, it would be better to construct this as an associative list. See https://hackmd.io/@tgstation/ry6DxGvDL#Associative-vs-Normal-Lists and https://www.byond.com/docs/ref/#/list/associations